src/cargo/sources/git/utils.rs

```diff
@@ -694,12 +695,40 @@ pub fn with_fetch_options(
     let mut progress = Progress::new("Fetch", config);
     network::with_retry(config, || {
         with_authentication(url, git_config, |f| {
+            let mut last_recv = 0.0; // in Byte
+            let mut last_rate = 0.0; // in Byte/s
+            let mut last_update = Instant::now();
             let mut rcb = git2::RemoteCallbacks::new();
             rcb.credentials(f);
-
             rcb.transfer_progress(|stats| {
+                let indexed_deltas = stats.indexed_deltas();
+                let msg = if indexed_deltas > 0 {
+                    // Resolving deltas.
+                    format!(" ({}/{})", indexed_deltas, stats.total_deltas())
+                } else {
+                    // Receiving objects.
+                    let duration = last_update.elapsed();
+                    let (recv, rate) = if duration > Duration::from_secs(1) {
```

Comment:
This felt a little odd to me in that `tick` I think has internal rate limiting so I don't think it's necessary to add more rate limiting to here. Perhaps this could use the 1s interval to reset the `last_*` variables but otherwise the numbers could still be updated at a rate perhaps larger than once per second?

Reply:
Got it. Will remove the rate limiting and calculate the average for last 5 seconds only for a more smooth data view.